src/renderer/services/midgard/actionsHistory.ts

```diff
@@ -0,0 +1,80 @@
+import * as RD from '@devexperts/remote-data-ts'
+import * as A from 'fp-ts/Array'
+import * as FP from 'fp-ts/function'
+import * as Rx from 'rxjs'
+import * as RxOp from 'rxjs/operators'
+
+import { LiveData, liveData } from '../../helpers/rx/liveData'
+import { triggerStream } from '../../helpers/stateHelper'
+import { DefaultApi } from '../../types/generated/midgard/apis'
+import { InlineResponse200 } from '../../types/generated/midgard/models'
+import { MAX_ITEMS_PER_PAGE } from '../const'
+import { ErrorId } from '../wallet/types'
+import { getRequestType, mapActionDTO } from './actionsHistory.utils'
+import { HistoryActionsPageLD, TxType } from './types'
+
+export type LoadActionsParams = {
+  page: number
+  address?: string
+  txid?: string
+  asset?: string
+  type?: TxType | 'ALL'
+}
+
+export const DEFAULT_ACTIONS_HISTORY_REQUEST_PARAMS: LoadActionsParams = {
+  page: 1
+}
+
+export const createActionsHistoryService = (
+  byzantine$: LiveData<Error, string>,
+  getMidgardDefaultApi: (basePath: string) => DefaultApi
+) => {
+  const midgardDefaultApi$ = FP.pipe(byzantine$, liveData.map(getMidgardDefaultApi), RxOp.shareReplay(1))
+
+  const { stream$: reloadActionsHistory$, trigger: reloadActionsHistory } = triggerStream()
+
+  const actions$ = ({ page, type, ...params }: LoadActionsParams): HistoryActionsPageLD =>
+    FP.pipe(
+      Rx.combineLatest([midgardDefaultApi$, reloadActionsHistory$]),
+      RxOp.map(([api]) => api),
+      liveData.mapLeft(() => ({
+        errorId: ErrorId.GET_ACTIONS,
+        msg: 'API is not available'
+      })),
+      liveData.chain((api) =>
+        FP.pipe(
+          api.getActions({
+            ...params,
+            type: getRequestType(type),
+            limit: MAX_ITEMS_PER_PAGE,
+            // UI pages starts with 1 but backend's with 0
+            offset: MAX_ITEMS_PER_PAGE * (page - 1)
```

Comment:
Please handle `(page - 1)` at view level, so you can pass here `page`. (we do it similar with tx history in asset detail view)